DigitalLearningSolutions.Data/Models/SessionData/Tracking/Deligate/Enrol/SessionEnrolDelegate.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace DigitalLearningSolutions.Data.Models.SessionData.Tracking.Delegate.Enrol
+{
+    using System;
+    public class SessionEnrolDelegate
+    {
+        public SessionEnrolDelegate()
+        {
+            Id = new Guid();
+        }
+        public Guid Id { get; set; }
+        public int? AssessmentID { get; set; }
```

Comment:
AssessmentID is used to store both Customisations.CustomisationID and SelfAssessments.ID. The difference is identified by IsSelfAssessment which is 0 for customisations and 1 for self assessments. The remainig logic of setting multi page form data, storing the object in session and then inserting the records in database remains same. While retreiving the supervisor roles, a check is performed to see if the activity relates to self assessment so that a decision can be made if roles need to be retrieved or not. Then finally data is saved either in Progress or candidateassessments table

Reply:
Sorry, Manish. Added that comment yesterday, found the answer this morning and forgot to remove the comment.